### PR TITLE
MockLogAppender takes string logger names for capturing

### PR DESCRIPTION
**Pull request by ywangd**
For classes that are not publically accessible, we can use the cannoical names for capturing.

### CODE REVIEW

File: modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java [hunk 1]
```diff
@@ -252,7 +252,7 @@ private void maybeLogForSuccessAfterRetries(int initialAttempt, String action) {
                 action,
                 blobStore.bucket(),
                 blobKey,
-                purpose,
+                purpose.getKey(),
```
Comment (ywangd) [inline modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3RetryingInputStream.java:255]: Irrelevant to this PR. But it's too trivial to have its own PR.

File: test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java [hunk 2]
```diff
@@ -241,8 +241,17 @@ public String toString() {
     }
 
     public Releasable capturing(Class<?>... classes) {
-        start();
         final var loggers = Arrays.stream(classes).map(LogManager::getLogger).toArray(Logger[]::new);
+        return appendToLoggers(loggers);
+    }
+
+    public Releasable capturing(String... names) {
+        final var loggers = Arrays.stream(names).map(LogManager::getLogger).toArray(Logger[]::new);
```
Comment (DaveCTurner) [inline test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java:249]: nit: not sure why I used `toArray` here in the first place, `toList()` would be neater

**Review by DaveCTurner — APPROVED** [review]
LGTM, one nit

File: test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java (hunk 2 above)
Comment (ywangd) [reply test/framework/src/main/java/org/elasticsearch/test/MockLogAppender.java]: Yep I updated both places to use `toList`.

**Pull request merged**